### DISCUSSION TITLE
Auto read OTP in Android using the Google SMS Verification API

### DISCUSSION THREAD

**Discussion by shery-sharaf** [dotnet/maui#16772] [Q&A]
Since READ_SMS and RECEIVE_SMS permissions are not so much easier to get published on the play store, how can i implement auto read OTP in android using the Google SMS verification API in .net MAUI. I came over this [Request SMS Verification in an Android App](https://developers.google.com/identity/sms-retriever/request). For implementing this how can i add the required dependencies of 'com.google.android.gms:play-services-auth:20.6.0' and 'com.google.android.gms:play-services-auth-api-phone:18.0.1' in .net MAUI ?

**Comment by themronion** [comment]
These packages are binded in C# and can be found on nuget. Search for them there

**Comment by eli191** [comment]
@shery-sharaf 
How have you done this please?
I get this error  `error CS0234: The type or namespace name 'Phone' does not exist in the namespace 'Android.Gms.Auth.Api'`
I have added the package Xamarin.GooglePlayServices.Base.

**Reply by LeoJHarris** [reply]
@eli191 did you add `Xamarin.GooglePlayServices.Auth.Api.Phone`
